security/fma-l1-pectra.md

```diff
@@ -0,0 +1,95 @@
+
+- [Github tracker](https://github.com/orgs/ethereum-optimism/projects/117/views/9)
+- [Specs clarification (Logs)](https://specs.optimism.io/protocol/derivation.html#on-future-proof-transaction-log-derivation)
+- [Specs clarification (L1 Retrieval)](https://specs.optimism.io/protocol/derivation.html#l1-retrieval)
+
+## Failure Modes and Recovery Paths
+
+### FM1: Consensus bug (client software)
+
+- **Description:** L2 EL/CL clients are not able to parse and/or validate blocks when Pectra goes live on L1, those nodes may halt. This could happen a) if operators do not update their nodes to a suitable release or release candidate before Pectra activates on L1 or b) no such release is available in time or c) there is a bug in such releases.
+- **Risk Assessment:** High severity, Low Likelihood
+- **Mitigations:**:
+  1. We will rely on unit tests, [end-to-end tests](https://github.com/ethereum-optimism/optimism/pull/14006) and cross-client devnet acceptance tests to ensure no client halts when L1 activates Pectra. In particular, we performed [manual Kurtosis testing](https://github.com/ethereum-optimism/optimism/pull/14046) with an L1 that has Pectra activated, and sent EIP-7702 transactions to the L1. This mitigates c.
+  2. Pectra will activate on Sepolia before L1 mainnet. This will provide another test (in production) of the node software changed. If the sepolia superchain continues to progress its safe chains, this will give us high confidence that the mainnet superchain will also. This mitigates b and c.
+  3. We will make our software releases well ahead of time and communicate clearly to operators about the need to upgrade. See these docs https://docs.optimism.io/notices/pectra-changes. This mitigates a and b.
+- **Detection:** Manual or (preferably) automated/scheduled testing.
+- **Recovery Path(s)**: The affected clients would need to be patched as soon as possible and new releases cut and notices to upgrade broadcast.
+
+### FM2: Consensus bug (proof system)
+
+- **Description:** If any fault proof program is unable parse and/or validate blocks when Pectra goes live on L1, it may be impossible to prove correct blocks and defend against malicious challenges. See FM1 for various scenarios which trigger this failure mode.
+- **Risk Assessment:** High severity, Low Likelihood
+- **Mitigations:**:
+  1. Our [end-to-end tests](https://github.com/ethereum-optimism/optimism/pull/14006) include coverage for `op-program` (this runs in CI) as well as `kona` (this has been run manually and passes).
+  2. If upstream work does not yet allow for appropriate end-to-end tests, we can patch our L1 clients in the testing environment(s) so we can still run the tests.
+- **Detection:** Automatic proofs monitoring systems would alert on-call engineers quickly in this instance. We run the op-challenger (with the new absolute prestate) on our production networks already, so we this part of the system will benefit from several weeks of battle-testing.
```

Comment:
> with the new absolute prestate

1. Where this Absolute Prestate is located (add link)
2. How to generate this Absolute Prestate (add link)
3. Is there is a way that this link will change from the dev to mainnet op-program is using the exact same input for the testnet so will produce the same Absolute prestate for mainnet in the future FDG?

Reply:
There is a good section on this in the optimism docs link that I added here 4dc65fe3702da3fee6424136320529cb328f6019. Hopefully that is sufficient!

As I understand it the new prestate mentioned in the docs works for various sepolia and mainnet OPStack chains. Outside of that set, a custom prestate should be generated (and this goes for devnets, too).